### PR TITLE
Improvement: [JW8-11087] Under specific scenarios enable native html5 element looping

### PR DESCRIPTION
**Pull request by imbcmdth**
### This PR will...
Set the `loop` attribute to `true` when:
1. The configuration has `repeat: true` 
2. There is only one item in the playlist
3. The item has a source that is a video of type `mp4`

### Why is this Pull Request needed?
Folks want to be able to use a short video that repeats and for the looping to be seamless.

### Are there any points in the code the reviewer needs to double check?
No

### Are there any Pull Requests open in other repos which need to be merged with this?
No

#### Addresses Issue(s):

JW8-11087

### Checklist
- [ ] Jenkins builds and unit tests are passing
- [ ] I have reviewed the automated results

### CODE REVIEW

File: src/js/api/config.js [hunk 1]
```diff
@@ -39,6 +39,8 @@ export const Defaults = {
     playbackRates: [0.5, 1, 1.25, 1.5, 2],
     renderCaptionsNatively: false,
     repeat: false,
+    // loop is only used on native html5 providers to loop mp4
+    loop: false,
```
Comment (zetagame) [inline src/js/api/config.js:43]: config is specific to anything a user can configure. You can store this attribute on the model without extending config.

File: src/js/controller/controller.js [hunk 2]
```diff
@@ -1310,6 +1310,16 @@ Object.assign(Controller.prototype, {
         } catch (error) {
             return Promise.reject(error);
         }
+
+        // Under very special conditions, we enable the html5 looping attribute
+        if (playlist.length === 1
```
Comment (zetagame) [inline src/js/controller/controller.js:1315]: Is it possible to do this in the HTML5 provider?

Comment (imbcmdth) [reply src/js/controller/controller.js:1315]: I was trying to keep the provider out of worrying about the playlist level info - to keep responsibilities separate. I know the model is available pretty much everywhere but it feels wrong. I'll try it and see what it's like.

Comment (zetagame) [reply src/js/controller/controller.js:1315]: The controller is pretty high level for something specific to mp4 sources, especially given the additional constraints (it's essentially for API-level functionality). HTML5 provider's setPlaylistItem function might be a good place to put it. When you say the using the model feels wrong, what do you mean?

**Comment by esaari** [comment]
test this please

**Review by zetagame — APPROVED** [review]

**Pull request merged**